mbuild/formats/hoomdxml.py

```diff
@@ -134,95 +135,99 @@ def write_hoomdxml(structure, filename, forcefield, box, ref_distance=1.0,
+                bond_types = np.zeros(len(bonds), dtype=int)
             else:
-                unique_bond_types = dict(enumerate(set([(round(bond.type.k,3),
-                                                         round(bond.type.req,3)) for bond in structure.bonds])))
-                unique_bond_types = {y:x for x,y in unique_bond_types.items()}
-                bond_types = [unique_bond_types[(round(bond.type.k,3),
-                                                 round(bond.type.req,3))] for bond in structure.bonds]
+                unique_bond_types = dict(enumerate(set([(round(bond.type.k, 3),
+                                                         round(bond.type.req, 3)) for bond in structure.bonds])))
+                unique_bond_types = {y: x for x, y in unique_bond_types.items()}
+                bond_types = [unique_bond_types[(round(bond.type.k, 3),
+                                                 round(bond.type.req, 3))] for bond in structure.bonds]
                 xml_file.write('<bond_coeffs>\n')
-                for params,idx in unique_bond_types.items():
-                    xml_file.write('{}\t{}\t{}\n'.format(idx,((params[0]*2.)/ref_energy)*(ref_distance**2.),params[1]/ref_distance))
+                for params, idx in unique_bond_types.items():
+                    xml_file.write('{}\t{}\t{}\n'.format(idx, ((params[0]*2.) / ref_energy)*(ref_distance**2.), params[1]/ref_distance))
                 xml_file.write('</bond_coeffs>\n')
             xml_file.write('<bond>\n')
-            for idx,bond in enumerate(bonds):
-                xml_file.write('{}\t{}\t{}\n'.format(bond_types[idx],*bond))
+            for idx, bond in enumerate(bonds):
+                xml_file.write('{}\t{}\t{}\n'.format(bond_types[idx], *bond))
             xml_file.write('</bond>\n')
 
-        angles = [[angle.atom1.idx, 
-                   angle.atom2.idx, 
+        angles = [[angle.atom1.idx,
+                   angle.atom2.idx,
                    angle.atom3.idx] for angle in structure.angles]
         if angles:
-            unique_angle_types = dict(enumerate(set([(round(angle.type.k,3), 
-                                                      round(angle.type.theteq,3)) for angle in structure.angles])))
-            unique_angle_types = {y:x for x,y in unique_angle_types.items()}
-            angle_types = [unique_angle_types[(round(angle.type.k,3), 
-                                               round(angle.type.theteq,3))] for angle in structure.angles]
+            unique_angle_types = dict(enumerate(set([(round(angle.type.k, 3),
+                                                      round(angle.type.theteq, 3)) for angle in structure.angles])))
+            unique_angle_types = {y: x for x, y in unique_angle_types.items()}
+            angle_types = [unique_angle_types[(round(angle.type.k, 3),
+                                               round(angle.type.theteq, 3))] for angle in structure.angles]
```

Comment:
I will dig up the code we use to see how it works. I know in our group we would consider the angle types to be different because even though theta and k were the same, the types were different. I'm not sure what the benefit is to considering them to be same. I know that would add verbosity to the force field file but since that is going to be created 'automatically' then I think its okay.